src/tespy/tools/logger.py

```diff
@@ -16,6 +16,259 @@
 
 import tespy
 
+TESPY_LOGGER_ID = "TESPyLogger"
+TESPY_PROGRESS_LOG_LEVEL = logging.DEBUG + 1 # 11
+TESPY_RESULT_LOG_LEVEL = logging.DEBUG + 2 # 12
+
+# Capture warnings globally instead of per file
+logging.captureWarnings(True)
+
+# Create a bunch of shorthand functions, this is mostly 
+# copied straight from the logging module.
+def getLogger():
+    return logging.getLogger(TESPY_LOGGER_ID)
+
+def log(level, msg, *args, **kwargs):
+    """
+    Log 'msg % args' with the integer severity 'level'.
+
+    To pass exception information, use the keyword argument exc_info with
+    a true value, e.g.
+
+    log(level, "We have a %s", "mysterious problem", exc_info=1)
+    """
+    logger = getLogger()
+    # We force the logging framework to trace past this file
+    if "stacklevel" not in kwargs:
+        kwargs["stacklevel"] = 1
+    kwargs["stacklevel"] += 1
+    # Last exit for Python < 3.8
+    if sys.version_info.major < 3 or (sys.version_info.major == 3 and sys.version_info.minor < 8):
+        kwargs.pop("stacklevel")
+    logger.log(level, msg, *args, **kwargs)
+
+def debug(msg, *args, **kwargs):
+    """
+    Log 'msg % args' with severity 'DEBUG'.
+
+    To pass exception information, use the keyword argument exc_info with
+    a true value, e.g.
+
+    debug("Houston, we have a %s", "thorny problem", exc_info=1)
+    """
+    # We force the logging framework to trace past this file
+    if "stacklevel" not in kwargs:
+        kwargs["stacklevel"] = 1
+    kwargs["stacklevel"] += 1
```

Comment:
This specific piece could be a function, since it is used in all logging functions. Or we can use `kwargs.get("stacklevel", 1) + 1`.

Reply:
I discovered that was incorrect, I implemented it however in an own function.

Reply:
Thank you for reviewing that part.